diff --git a/gcc/haifa-sched.c b/gcc/haifa-sched.c
--- a/gcc/haifa-sched.c
+++ b/gcc/haifa-sched.c
@@ -3869,16 +3869,11 @@
 		  {
 		    for (u = reg_last_uses[i]; u; u = XEXP (u, 1))
 		      add_dependence (insn, XEXP (u, 0), REG_DEP_ANTI);
-		    reg_last_uses[i] = 0;
 
 		    for (u = reg_last_sets[i]; u; u = XEXP (u, 1))
 		      add_dependence (insn, XEXP (u, 0), REG_DEP_ANTI);
 
-		    if (global_regs[i])
-		      for (u = reg_last_clobbers[i]; u; u = XEXP (u, 1))
-		        add_dependence (insn, XEXP (u, 0), REG_DEP_ANTI);
-
-		    SET_REGNO_REG_SET (reg_pending_sets, i);
+		    SET_REGNO_REG_SET (reg_pending_clobbers, i);
 		  }
 	    }
 
